analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -64,6 +65,7 @@ protected void ReadParameters(SonarAnalysisContext context, CompilationAnalysisC
                 AnalyzeGeneratedCode = PropertiesHelper.ReadAnalyzeGeneratedCodeProperty(settings, c.Compilation.Language);
                 OutPath = Path.Combine(outPath, c.Compilation.Language == LanguageNames.CSharp ? "output-cs" : "output-vbnet");
                 IsAnalyzerEnabled = true;
+                IsTestProject = context.IsTestProject(c.Compilation, c.Options);
```

Comment:
this change also needs an integration test - that's the only way to verify it E2E

Reply:
as discussed offline, can be done separately as this PR doesn't change the functional behavior